nautobot/core/views/mixins.py

```diff
@@ -499,9 +500,19 @@ def get_queryset(self):
         queryset = super().get_queryset()
         return queryset.restrict(self.request.user, self.get_action())
 
+    def get_action_map(self):
+        """
+        Return a mapping of action names to their corresponding permissions.
+        This is used to determine the permissions required for each action.
+        """
+        if self.custom_action_permission_map:
+            return dict(self.custom_action_permission_map, **PERMISSIONS_ACTION_MAP)
```

Comment:
I forget - which one "wins" if there's an overlap between the two? Does this allow for, say,`custom_action_permission_map = {"retrieve": "change"}` or no?